### PR TITLE
Enhancements and fixes

### LINKED ISSUE #312: add destination_site to Filestreams::Image Solr records
To support page-level search results in a newspaper content display system, we will need to add a `destination_site_ssim` value to the Solr record for the `Filestreams::Image` object representing a newspaper page.

This field should ONLY be added when the parent `DigitalObject`'s `Metastreams::Administrative#destination_site` value includes `newspapers`. The value that should be added is `newspapers`.

### LINKED ISSUE #335: don't add "Serial Publications" genre facet to Curator::Collection
In [Curator::CollectionIndexer](https://github.com/boston-library/curator/blob/master/app/indexers/curator/collection_indexer.rb#L19-L30), we should exclude `Serial Publications` from the `genre_labels` list that gets indexed.

That value is only for newspaper "title-level" objects, which are not intended to be returned in regular search results.

### LINKED ISSUE #358: Updating institution using `InstitutionUpdaterService` breaks unless location is in the params

### LINKED ISSUE #359: create scope called `issue_objects` for `DigitalObject`
the scope should just be 

```
 scope :issue_objects -> { where.not(contained_by_id: nil) }
```

### LINKED ISSUE #360: add has_transcription_bsi to DigitalObjectIndexer
In order to indicate when an item has an associated text transcription, we should add a field `has_transcription_bsi` to `DigitalObjectIndexer`.

(This is different than the existing `has_searchable_pages_bsi` field, since some items, like PDFs and AV materials may have an associated transcription without having associated individual page images.)

Change this:
```ruby
text_plain_attachment = record.text_file_sets.first.text_plain_attachment

next if text_plain_attachment.blank?

text_plain_attachment.download do |file|
  context.output_hash['ocr_tiv'] = Curator::Parsers::InputParser.clean_text(file)
end
```
to:
```ruby
text_plain_attachment = record.text_file_sets.first.text_plain_attachment
next if text_plain_attachment.blank?

context.output_hash['has_transcription_bsi'] = true
text_plain_attachment.download do |file|
  context.output_hash['ocr_tiv'] = Curator::Parsers::InputParser.clean_text(file)
end
```

Once this is added, we should iterate over all `Filestreams::Text` objects and reindex the associated `DigitalObject`.

### PR DESCRIPTION
**Pull request by bbarberBPL**
- Made all `solr_wrapper` config files use `8.11.3` as the version

- Added `has_trasncription_bsi` to `digital_object` solr record if `ocr_tiv` is present, resolves #360

- Use safe operator on location variable for institution updater service, fixes #358

- Created `issue_objects` and `with_admin_set_ark` scopes on digital objects for easier querying, resolves #359

- Prevent `Serial Publications` from being added to genre facet on collections indexer, resolves #335

- Added newspapers to the `destination_site_ssim` field of file sets that are images and with parents that have newspapers in the same field, resolves #312

### CODE REVIEW

File: app/models/curator/digital_object.rb [hunk 1]
```diff
@@ -15,6 +15,9 @@ class DigitalObject < ApplicationRecord
 
     scope :for_serialization, -> { includes(:file_sets, exemplary_image_mapping: :exemplary_file_set).with_metastreams }
     scope :for_reindex_all, -> { for_serialization.joins(:administrative, :descriptive, :workflow) }
+    scope :issue_object, -> { where.not(contained_by_id: nil) }
```
Comment (ebenenglish) [inline app/models/curator/digital_object.rb:18]: Do we want to call this `issue_objects` as in #359?

File: app/models/curator/digital_object.rb [hunk 2]
```diff
@@ -15,6 +15,9 @@ class DigitalObject < ApplicationRecord
...
+    scope :with_admin_set_ark, ->(admin_set_ark_id) { joins(:admin_set).where(admin_set: { ark_id: admin_set_ark_id }) }
```
Comment (ebenenglish) [inline app/models/curator/digital_object.rb:19]: Which issue is this line intended to address? Or is it just for convenience?

File: spec/models/curator/digital_object_spec.rb [hunk 3]
```diff
@@ -163,6 +163,31 @@
       end
     end
 
+    describe '.issue_object' do
```
Comment (ebenenglish) [inline spec/models/curator/digital_object_spec.rb:166]: Same comment as above, should this be `issue_objects`?

**Review by ebenenglish — COMMENTED** [review]
Looks good overall, couple of questions.

File: app/indexers/curator/digital_object_indexer.rb [hunk 4]
```diff
@@ -16,11 +16,14 @@ class DigitalObjectIndexer < Curator::Indexer
...
         accumulator.concat record.is_member_of_collection.pluck(:name)
       end
+
       to_field 'collection_ark_id_ssim' do |record, accumulator|
         accumulator.concat record.is_member_of_collection.pluck(:ark_id)
       end
+
       to_field 'contained_by_ssi', obj_extract('contained_by', 'ark_id')
-      to_field('filenames_ssim') { |rec, acc| acc.concat rec.file_set_members.pluck(:file_name_base).uniq }
+      to_field('filenames_ssim') { |rec, acc| acc.concat rec.file_set_members.distinct.pluck(:file_name_base) }
```
[the remark this answers is not in the document]
Comment (bbarberBPL) [reply app/indexers/curator/digital_object_indexer.rb:25]: doing `.distinct.pluck(...)` does the sql equivalent of that

File: app/models/curator/digital_object.rb (hunk 2 above)
Comment (bbarberBPL) [reply app/models/curator/digital_object.rb:19]: Just convenience. Mostly for querying newspaper objects stuck in derivatives

**Review by ebenenglish — APPROVED** [review]

**Pull request merged**